File: ai-job-portal-frontend/src/features/recruiter-dashboard/services/recruiter-dashboard.service.ts
import { apiClient } from "@/services/api-client";
import type { ApiResponse } from "@/types/api";
import type { RecruiterDashboardResponse } from "@/features/recruiter-dashboard/types";

/** Maps 1:1 to recruiter-service's RecruiterDashboardController. */
export const recruiterDashboardService = {
  getDashboard: () =>
    apiClient.get<ApiResponse<RecruiterDashboardResponse>>("/recruiter/dashboard").then((res) => res.data),
};

File: ai-job-portal-backend/recruiter-service/src/main/java/com/prahlad/aijobportal/recruiterservice/dashboard/service/RecruiterDashboardService.java
package com.prahlad.aijobportal.recruiterservice.dashboard.service;

import com.prahlad.aijobportal.recruiterservice.dashboard.dto.response.RecruiterDashboardResponse;

/**
 * DAY11 "Recruiter Dashboard Improvements". Extends recruiter-service
 * only (no new microservice) and aggregates read-only data already
 * owned by Job Service, Application Service, and AI Service via Feign
 * — it never duplicates or persists their data locally.
 */
public interface RecruiterDashboardService {

    RecruiterDashboardResponse getDashboard(String bearerToken);
}
